glean-core/src/lib.rs

```diff
@@ -285,6 +298,12 @@ pub trait OnGleanEvents: Send {
     }
 }
 
+/// A callback handler that receives the IDs of recorded events
+pub trait GleanEventListener: Send {
+    /// Called when an event is recorded, indicating the id of the event
+    fn on_event_recorded(&self, id: String);
```

Comment:
Why `String` and not `&str`? I don't imagine listeners will want to own the FQMN if they can avoid it?

Reply:
I don't think `&str` is possible through UniFFI...  and the metric `base_identifier()` function returns a `String` rather than `&str`.

Nimbus is planning on using this value as a key lookup to increment counters in a map structure, so their "ownership" of this value will be quickly garbage collected, anyway.